nibabel/filebasedimages.py

```diff
@@ -47,19 +52,19 @@ def from_header(klass, header=None):
         )
 
     @classmethod
-    def from_fileobj(klass, fileobj: io.IOBase):
-        raise NotImplementedError
+    def from_fileobj(klass: type[HdrT], fileobj: io.IOBase) -> HdrT:
+        raise NotImplementedError  # pragma: no cover
 
-    def write_to(self, fileobj: io.IOBase):
-        raise NotImplementedError
+    def write_to(self, fileobj: io.IOBase) -> None:
+        raise NotImplementedError  # pragma: no cover
 
-    def __eq__(self, other):
-        raise NotImplementedError
+    def __eq__(self, other: object) -> bool:
+        raise NotImplementedError  # pragma: no cover
 
-    def __ne__(self, other):
+    def __ne__(self, other: object) -> bool:
         return not self == other
 
-    def copy(self) -> FileBasedHeader:
+    def copy(self: HdrT) -> HdrT:
```

Comment:
The idea here is that you have a `.copy()` being called on a `FileBasedHeader` instance, which will almost always be some subclass, and return an instance of the same type. To assert that the result is not just a `FileBasedHeader` but the same subclass of `FileBasedHeader`, we use a type variable bound to `FileBasedHeader`.

As it was, `Nifti1Header().copy().get_sform()` would not type check. Now it will.